pkg/apis/v1/nodeclaim.go

```diff
@@ -75,6 +75,78 @@ type NodeClaimSpec struct {
 	ExpireAfter NillableDuration `json:"expireAfter,omitempty"`
 }
 
+func (in *NodeClaimSpecWithoutResources) ToNodeClaimSpec() *NodeClaimSpec {
+	return &NodeClaimSpec{
+		Taints:                 in.Taints,
+		StartupTaints:          in.StartupTaints,
+		Requirements:           in.Requirements,
+		NodeClassRef:           in.NodeClassRef,
+		TerminationGracePeriod: in.TerminationGracePeriod,
+		ExpireAfter:            in.ExpireAfter,
+	}
+}
+
+func (in *NodeClaimSpec) ToNodeClaimSpecWithoutResources() *NodeClaimSpecWithoutResources {
+	return &NodeClaimSpecWithoutResources{
+		Taints:                 in.Taints,
+		StartupTaints:          in.StartupTaints,
+		Requirements:           in.Requirements,
+		NodeClassRef:           in.NodeClassRef,
+		TerminationGracePeriod: in.TerminationGracePeriod,
+		ExpireAfter:            in.ExpireAfter,
+	}
+}
+
+// NodeClaimSpec describes the desired state of the NodeClaim
+type NodeClaimSpecWithoutResources struct {
```

Comment:
Consider calling this NodeClaimTemplateSpec and storing this in the NodePool.go file

Reply:
Good call